src/pages/scan/scan.ts

```diff
@@ -141,3 +147,53 @@ export async function handleParticipant(
     handleError(response, 'Could not fetch participant data');
   }
 }
+
+interface CustomCodeHandlerWithBaseUrl {
+  baseUrl: string;
+  name: string;
+  regex: string;
+}
+
+export async function parseCustomQRCodeData(
+  decodedText: string,
+  errorModal: ErrorModalFunction
+): Promise<IndicoParticipant | null> {
+  const availableServers = await getServers();
+  const customCodeHandlers: CustomCodeHandlerWithBaseUrl[] = availableServers.flatMap(server =>
+    server.customCodeHandlers
+      ? Object.entries(server.customCodeHandlers).map(([name, regex]) => ({
+          baseUrl: server.baseUrl,
+          name,
+          regex,
+        }))
+      : []
+  );
+
+  for (const customCodeHandler of customCodeHandlers) {
+    let regex;
+    try {
+      regex = new RegExp(customCodeHandler.regex);
+    } catch {
+      return null;
+    }
+    if (regex.test(decodedText)) {
+      const server = await db.servers.get({baseUrl: customCodeHandler.baseUrl});
+      if (!server) {
```

Comment:
This can no longer happen, can it?

It might be nicer to avoid creating the the `customCodeHandlers` list above and just iterate over the servers and their custom code handlers, then you already have the server available and don't need to look it up again.